### PR TITLE
Add COOP & COEP headers for SvelteKit, NextJS, & NuxtJS

### PR DESCRIPTION
**Pull request by jasongitmail**
My first commit covers Svelte Kit and NextJS. NuxtJS is still todo.

Closes https://github.com/o1-labs/zkapp-cli/issues/267

### CODE REVIEW

**Review by MartinMinkov — APPROVED** [review]
Awesome! Nuxt will be coming up soon!

**Review by ymekuria — APPROVED** [review]
LGTM! Exciting!

**Comment by jasongitmail** [comment]
@ymekuria Martin added code for Nuxt and updated the Next code. Would you mind try it out again please?

File: src/lib/project.js [hunk 1]
```diff
@@ -286,6 +273,78 @@ function kebabCase(str) {
...
+  );
+}
+
+function scaffoldNext() {
+  // https://nextjs.org/docs/api-reference/create-next-app#options
+  spawnSync('npx', ['create-next-app@latest', 'ui', '--use-npm'], {
+    stdio: 'inherit',
+    shell: true,
+  });
+  sh.rm('-rf', path.join('ui', '.git')); // Remove NextJS' .git; we will init .git in our monorepo's root.
```
Comment (ymekuria) [inline src/lib/project.js:294]: Are we init .git in the mono repo yet?

File: src/lib/project.js [hunk 2]
```diff
@@ -286,6 +273,78 @@ function kebabCase(str) {
   return str.toLowerCase().replace(' ', '-');
 }
 
+function scaffoldSvelte() {
```
Comment (ymekuria) [inline src/lib/project.js:276]: Abstracting the scaffolding into the functions is a clean approach.

File: src/lib/project.js [hunk 3]
```diff
@@ -286,6 +273,78 @@ function kebabCase(str) {
...
+  spawnSync('npx', ['create-next-app@latest', 'ui', '--use-npm'], {
+    stdio: 'inherit',
+    shell: true,
+  });
+  sh.rm('-rf', path.join('ui', '.git')); // Remove NextJS' .git; we will init .git in our monorepo's root.
+  // Read in the NextJS config file and add the middleware.
+  const nextConfig = fs.readFileSync(path.join('ui', 'next.config.js'), 'utf8');
+  const newNextConfig = nextConfig.replace(
+    /^}(.*?)$/gm, // Search for the last '}' in the file.
+    `
```
Comment (ymekuria) [inline src/lib/project.js:299]: If the standard generated nextConfig is:

```
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  swcMinify: true,
}

module.exports = nextConfig
```

would this `const newNextConfig = nextConfig.replace(
    /^}(.*?)$/gm, // Search for the last '}' in the file.` remove the `}` under the `swcMinify: true,` line - ie the closing bracket of the nextConfig object, or am I reading this wrong?

**Review by ymekuria — APPROVED** [review]

**Comment by ymekuria** [comment]
I was able to generate ui projects that had the correct headers with all the supported frameworks.

**Pull request merged**